app/gui/src/dashboard/modals/DuplicateAssetsModal.tsx

```diff
@@ -316,3 +339,365 @@ export default function DuplicateAssetsModal(props: DuplicateAssetsModalProps) {
     </Modal>
   )
 }
+
+/**
+ * The conclusion of a resolved duplication.
+ */
+export type Conclusion = 'rename' | 'replace' | 'skip'
+
+/**
+ * A resolved duplication.
+ */
+export type ResolvedDuplication = RenameDuplication | ReplaceDuplication | SkipDuplication
+
+/**
+ * A resolved duplication that was skipped.
+ */
+export interface SkipDuplication {
+  readonly assetId: backendModule.AssetId
+  readonly conclusion: 'skip'
+}
+
+/**
+ * A resolved duplication that was renamed.
+ */
+export interface RenameDuplication {
+  readonly assetId: backendModule.AssetId
+  readonly conclusion: 'rename'
+  readonly newName: string
+}
+
+/**
+ * A resolved duplication that was replaced.
+ */
+export interface ReplaceDuplication {
+  readonly assetId: backendModule.AssetId
+  /**
+   * Requires backend to support that.
+   */
+  readonly conclusion: 'replace'
+}
+
+/**
+ * Props for a {@link ResolveDuplicationsModal}.
+ */
+export interface ResolveDuplicationsProps {
+  readonly targetId: backendModule.DirectoryId
+  readonly conflictingIds: readonly backendModule.AssetId[]
+  readonly onSubmit: (assets: readonly ResolvedDuplication[]) => Promise<void> | void
+  readonly onCancel: () => void
+}
+
+/**
+ * A modal for resolving duplicates.
+ */
+export function ResolveDuplicationsModal(props: ResolveDuplicationsProps) {
+  const { conflictingIds } = props
+  const { getText } = textProvider.useText()
+
+  return (
+    <Dialog
+      size="xxlarge"
+      onDismiss={props.onCancel}
+      title={
+        conflictingIds.length === 1 ?
+          getText('resolveDuplicatesTitleOne')
+        : getText('resolveDuplicatesTitleMany', conflictingIds.length)
+      }
+    >
+      <ResolveDuplicationsModalInner {...props} />
+    </Dialog>
+  )
+}
+
+const NEW_TITLE_SUFFIX = ' (copy)'
+
+/**
+ * The inner component of a {@link ResolveDuplicationsModal}.
+ */
+function ResolveDuplicationsModalInner(props: ResolveDuplicationsProps) {
```

Comment:
consider renaming to `ResolveDuplicationsForm` - some of our forms aren't always in a modal

Reply:
Well, I’m not certain if this component has any practical use cases for being inline. I think we can consider that for later.
It’s worth noting that this component was extracted solely to isolate suspense and error boundaries within the dialog.